curator-framework/src/main/java/org/apache/curator/framework/state/ConnectionStateManager.java

```diff
@@ -252,7 +252,9 @@ private void processEvents()
             {
                 int lastNegotiatedSessionTimeoutMs = client.getZookeeperClient().getLastNegotiatedSessionTimeoutMs();
                 int useSessionTimeoutMs = (lastNegotiatedSessionTimeoutMs > 0) ? lastNegotiatedSessionTimeoutMs : sessionTimeoutMs;
-                int pollMaxMs = (useSessionTimeoutMs * 2) / 3; // 2/3 of session timeout
+                long elapsedMs = startOfSuspendedEpoch == 0 ? 0 : System.currentTimeMillis() - startOfSuspendedEpoch;
+                long pollMaxMs = useSessionTimeoutMs - elapsedMs;
```

Comment:
What do you mean if there isn't a suspended state? Shouldn't we go into a suspended state as soon as the connection is dropped? Then, we will wait up to session timeout for an event of any type. If we don't get one, then we assume the session is lost and inject a LOST event.